core/src/main/java/org/elasticsearch/search/highlight/HighlightBuilder.java

```diff
@@ -378,9 +379,15 @@ public SearchContextHighlight build(QueryShardContext context) throws IOExceptio
      */
     @SuppressWarnings({ "rawtypes", "unchecked" })
     private static void transferOptions(AbstractHighlighterBuilder highlighterBuilder, SearchContextHighlight.FieldOptions.Builder targetOptionsBuilder, QueryShardContext context) throws IOException {
-        targetOptionsBuilder.preTags(highlighterBuilder.preTags);
-        targetOptionsBuilder.postTags(highlighterBuilder.postTags);
-        targetOptionsBuilder.scoreOrdered("score".equals(highlighterBuilder.order));
+        if (highlighterBuilder.preTags != null) {
+            targetOptionsBuilder.preTags(highlighterBuilder.preTags);
+        }
+        if (highlighterBuilder.postTags != null) {
+            targetOptionsBuilder.postTags(highlighterBuilder.postTags);
+        }
+        if (highlighterBuilder.order != null) {
+            targetOptionsBuilder.scoreOrdered("score".equals(highlighterBuilder.order));
```

Comment:
maybe we should convert this to use ParseField so that we can use the parseFieldMatcher here in case we end up with deprecations in the order names?